ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -883,20 +1032,42 @@ export const RuntimePanel = fp.flow(
   if (currentRuntime === undefined) {
     return <Spinner style={{width: '100%', marginTop: '5rem'}}/>;
   }
-
+  const createStandardComputeRuntimeRequest = (runtime: RuntimeConfig) => {
+    // The logic here is tricky to be compatible
+    // Use gceConfig when PD feature is not enabled OR reduce PD size OR update an existing runtime with no PD attached.
+    // TODO(RW-): 'Update an existing runtime with no PD attached' will be cleaned up
+    // post launch PD when all existing running Runtime shutdown.
+    if (!enablePD || (!pdSizeReduced && runtimeExists) || runtimeExistsWithoutPD) {
+      return {
+        gceConfig: {
+          machineType: runtime.machine.name,
+          diskSize: runtime.diskSize
+        }
+      };
+    } else {
+      return {
+        gceWithPdConfig: {
+          bootDiskSize: 50,
+          machineType: runtime.machine.name,
+          persistentDisk: {
+            // When reducing PD size, passing empty name to backend then API will create a new PD
+            name: !pdExists || pdSizeReduced ? '' : persistentDisk.name,
```

Comment:
do we need `!pdExists`? If pd not exist, would persistentDisk.name is just empty?

Reply:
If pd not exist, persistentDisk will be null and throw NPE.

Reply:
We can make persistentDisk.name return empty string when null in diskStore

Reply:
But that still can't fully replace this logic here when reducing the pd size